reroll-initiative/old/reroll-initiative.js

```diff
@@ -0,0 +1,260 @@
+            settings[setting] = value;
+            await game.settings.set(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key,settings);
+        }
+        else{
+            console.error("Module setting : "+setting+" does not exist!");
+        }
+    }
+
+    /**
+     * Helper function to retrieve module settings from game instance
+     * @returns {Object} settings An object containing the queried settings
+     */
+    static querySettings() {
+        return game.settings.get(RerollInitiative.SETTINGS.module,RerollInitiative.SETTINGS.key)
+    }
+
+    /**
+     * Hook on combat update and if round in update is greater than previous -- call resetAndReroll
+     */
+    _postUpdateCombatHook() {
+        Hooks.on("updateCombat", (combat,update) =>  {
+            //this is probably an unnecessary hit on the db
+            //this._loadSettings();
+
+            if(this.config.reroll){
+                
+                if(update.round && combat.previous && update.round > combat.previous.round){
+                    //console.log("Reroll-Initiative: Round incremented - rerolling initiative")
+                    this._resetAndReroll(combat);
+                }
+            }
+            else {
+                console.log("Rerolling Initiative is currently turned off")
+            }
+            
+        }); 
+    }
+
+    /**
+     * For the given combat instance, call the resetAll method and the rollAll method
+     * @param {Object} combat A Combat instance
+     */
+    async _resetAndReroll(combat){
+        await combat.resetAll();
+        combat.rollAll();
+    }
+}
+
+/**
+ * @class RerollInitiativeConfig
+ * @description Handles the configuration form for the module. Currently inserts within Combat Tracker Config
+ * @todo: tear down all the setting updates here and use the RerollInitiative class instead
+ */
+class RerollInitiativeConfig {
+
+    constructor(){
+        this._hookRenderCombatTrackerConfig();
+    }    
+
+    /**
+     * Hooks on the render of combat tracker config and insert the module config
+     */
+    _hookRenderCombatTrackerConfig(){
+        Hooks.on("renderCombatTrackerConfig", (app, html) => {
+            
+            let settings = RerollInitiative.querySettings();
+            console.log(settings);
+            const targetSetting = "reroll";
+
+            if(html){
+                let rriCheckbox = this._injectCheckboxFormgroup(html);
+
+                if(rriCheckbox){
+                    //Set the state of the checkbox to match the current module reroll setting
+                    rriCheckbox.prop("checked",settings.reroll);
+
+                    // Adjust the window height to allow for new content
+                    app.setPosition({height: app.position.height + 60});
+                
+                    // Handle form submission
+                    //todo: break this out into a separate method?
+                    const submit = html.find('button[type="submit"]');
```

Comment:
My previously comment is still valid here. Finding by this is usually to generic. If there are multiple submit buttons on the page, you could potentially grab the wrong one. You should look at giving your button a unique ID and doing he lookup via that id